src/auth/auth.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { LoginUserDto } from 'src/auth/dto/login-user.dto';
+import { AccessToken } from './interfaces/accessToken';
+import { UsersService } from 'src/users/users.service';
+import * as bcrypt from 'bcrypt';
+import { JwtService } from '@nestjs/jwt';
+import { PrismaService } from 'src/prisma.service';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private usersService: UsersService,
+    private jwtService: JwtService,
+    private prisma: PrismaService
+  ) {}
+
+
+  private async validateEmail(email: string) {
+    // TODO: check that such email address doesn't already exist in the db,
+    // send a validation email to this address
+  }
+
+  // Sign user in by email and password
+  async signIn(credentials: LoginUserDto): Promise<AccessToken> {
+    const user = await this.prisma.user.findUnique({
+      where: {email: credentials.email}
+    });
+    await this.usersService.findOne({email: credentials.email});
```

Comment:
`AuthService` does not have responsibility to use `prisma.user`, as it *should* be reserved to `UsersService`. I recommend using a findOne-like.

Note: as `findOne` is returning a `SanitizedUser`, you could use a `findOneUnsanitized` that would return the full user, which would include the password.

```suggestion
    const user = await this.usersService.findOne({email: credentials.email});
```

Reply:
Ok, I'll add an `findOneUnsanitized` method into userService then.